lib/string_reverse.rb

```diff
@@ -1,6 +1,46 @@
+require "pry"
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string passed into the string_reverse method.

```

Comment:
_Time_ complexity LGTM. _Space_ complexity seems **higher** than `O(1)` as-is due to the `new_string` variable. (`O(1)` is probably achievable if we get rid of it).